### PR TITLE
[DRAFT] CB-5121 Hybrid environment

### PR DESCRIPTION
**Pull request by pkedvessy**
**DELETE THIS TEMPLATE BEFORE SUBMITTING**

Describe the change you are making here!

Please include tests. Check out these examples:

- https://github.com/hortonworks/cloudbreak/blob/master/core/src/test/java/com/sequenceiq/cloudbreak/service/blueprint/BlueprintLoaderServiceTest.java#L62-L76

Please include a short description. Check out these examples:

- https://github.com/hortonworks/cloudbreak/commit/888b2e2a8887def28a24e26efcd6d9ca5863733a

If your PR resolves any open issue(s), please indicate them like this so they will be closed when your PR is merged:

Closes #xxx
Closes #xxx

### CODE REVIEW

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java [hunk 1]
```diff
@@ -114,12 +115,17 @@ public FreeIpaCreationHandler(
...
         try {
-            if (environmentOptional.isPresent() && environmentOptional.get().isCreateFreeIpa()) {
+            if (environmentOptional.isPresent()) {
                 Environment environment = environmentOptional.get();
-                if (supportedPlatforms.supportedPlatformForFreeIpa(environment.getCloudPlatform())) {
-                    createFreeIpa(environmentDtoEvent, environmentDto, environment);
+                if (Objects.nonNull(environment.getParentEnvironment())) {
+                    attachParentFreeIpaToChild(environmentDtoEvent, environmentDto);
+                } else if (environmentOptional.get().isCreateFreeIpa()) {
+                    if (supportedPlatforms.supportedPlatformForFreeIpa(environment.getCloudPlatform())) {
```
Comment (lajosrodek) [inline environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java]: Not a big deal, but the nested `if` block could be merged into the enclosing one using a compound condition `...isCreateFreeIpa() && ...supportedPlatformForFreeIpa(...)`.

File: environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java [hunk 2]
```diff
@@ -67,6 +68,23 @@ public ValidationResultBuilder validateNetworkCreation(Environment environment,
...
     }
 
+    public ValidationResult validateParentChildRelation(Environment environment, String parentEnvironmentCrn) {
+        ValidationResultBuilder resultBuilder = new ValidationResultBuilder();
+
+        resultBuilder.ifError(() -> Objects.nonNull(parentEnvironmentCrn) && Objects.isNull(environment.getParentEnvironment()),
+                String.format("Active parent environment with crn '%s' is not available in account '%s'.", parentEnvironmentCrn, environment.getAccountId()));
+        if (Objects.nonNull(environment.getParentEnvironment())) {
+            resultBuilder.ifError(() -> Objects.nonNull(environment.getParentEnvironment().getParentEnvironment()),
+        "Parent environment is already a child environment.");
```
Comment (lajosrodek) [inline environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java]: Wrong indentation.

File: environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java (hunk 1 above)
Comment (pkedvessy) [reply environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/freeipa/FreeIpaCreationHandler.java]: Thank you. Fixed

File: environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java (hunk 2 above)
Comment (pkedvessy) [reply environment/src/main/java/com/sequenceiq/environment/environment/validation/EnvironmentValidatorService.java]: Thank you. Fixed

**Pull request closed without merging**